src/status_im/ui/screens/home/views.cljs

```diff
@@ -21,30 +21,72 @@
             [status-im.ui.components.colors :as colors])
   (:require-macros [status-im.utils.views :as views]))
 
+(defn welcome-image-wrapper []
+  (let [dimensions (reagent/atom {})]
+    (fn []
+      [react/view {:on-layout  (fn [e]
+                                 (reset! dimensions (js->clj (-> e .-nativeEvent .-layout) :keywordize-keys true)))
+                   :style {:align-items :center
+                           :justify-content :center
+                           :flex 1}}
+       (let [padding    0
+             image-size (- (min (:width @dimensions) (:height @dimensions)) padding)]
+         [react/image {:source (resources/get-image :welcome)
+                       :resize-mode :contain
+                       :style {:width image-size :height image-size}}])])))
+
 (defn welcome []
-  [react/view {:style styles/welcome-view}
-   [react/view {:flex 1}]
-   [status-bar/status-bar {:type :main}]
-   [react/view {:style styles/welcome-image-container}
-    [components.common/image-contain
-     {:container-style {}}
-     {:image (resources/get-image :welcome-image) :width 750 :height 556}]]
-   [react/i18n-text {:style styles/welcome-text :key :welcome-to-status}]
-   [react/view
-    [react/i18n-text {:style styles/welcome-text-description
-                      :key   :welcome-to-status-description}]]
-   [react/view {:flex 1}]
-   [react/view {:align-items :center :margin-bottom 52}
-    [components.common/button {:on-press #(re-frame/dispatch [:navigate-back])
-                               :label    (i18n/label :t/get-started)}]]])
+  [react/view {:style {:flex 1}}
+   [react/view {:style styles/welcome-view}
+    [status-bar/status-bar {:type :main}]
+    [welcome-image-wrapper]
+    [react/i18n-text {:style styles/welcome-text :key :welcome-to-status}]
+    [react/view
+     [react/i18n-text {:style styles/welcome-text-description
+                       :key   :welcome-to-status-description}]]
+    [react/view {:align-items :center :margin-bottom 50}
+     [components.common/button {:on-press #(re-frame/dispatch [:navigate-back])
+                                :label    (i18n/label :t/lets-go)}]]]])
+
+(defn- start-chat [topic]
+  (re-frame/dispatch [:chat.ui/start-public-chat topic {:navigation-reset? true}])
+  (re-frame/dispatch [:set :public-group-topic nil]))
+
+(defn chat-tags-view []
+  [react/view {:align-items :center :margin-top 16}
+   [react/i18n-text {:style styles/no-chats-text :key :follow-your-interests}]
+   [react/view {:style styles/tags-wrapper}
+    (for [item ["introductions", "chitchat", "status", "crypto", "tech", "music", "movies"]]
+      [react/touchable-highlight {:key item :on-press #(start-chat item)}
+       [react/text {:style styles/tag-text} (str "#" item)]])]])
+
+(def empty-template-visible? (reagent/atom true))
```

Comment:
Is it related with `hide-home-tooltip?` you mentioned above? I checked codes about `wallet-set-up-passed?`, does it mean that the value of `hide-home-tooltip?` should be stored on the server like `multiaccount-update` function?

Reply:
is the meaning of 'it should be defined in the view' like this? 
`(defn home-tooltip-view [blank?]
    (let [empty-template-visible? (reagent/atom true)]
    ...
    (if @empty-template-visible?
    ...
    [react/touchable-highlight {:on-press #(reset! empty-template-visible? false)}
   ...)`
But, it's not working properly to hide. 
Sorry to I'm clojure beginner, it would be appreciate you can advise.

Reply:
@elegant651

You need a reagent form-2 component https://github.com/reagent-project/reagent/blob/master/doc/CreatingReagentComponents.md#form-2--a-function-returning-a-function

Form 2 components are functions returning a function, the outer function is called when your component is used the first time, and the inner function is the react renderer. That way when your component is called, it initializes the atom, and it does not re-initialize it every time the renderer is called.
 
```clojure
(defn home-tooltip-view [blank?]
    (let [empty-template-visible? (reagent/atom true)]
    ...
    (fn [blank?]
    (if @empty-template-visible?
    ...
    [react/touchable-highlight {:on-press #(reset! empty-template-visible? false)}
   ...))
```

Reply:
> Is it related with hide-home-tooltip? you mentioned above? I checked codes about wallet-set-up-passed?, does it mean that the value of hide-home-tooltip? should be stored on the server like multiaccount-update function?

yes it is related you don't actually need an atom here because you should rely on the value of `hide-home-tooltip?` here to show the tooltip or not. And yes you should pretty much mimic the way `wallet-set-up-passed?` is used. These are not stored on the server though, they are stored in status-go which is running on the phone as well.